src/backend/distributed/commands/trigger.c

```diff
@@ -183,7 +198,22 @@ GetExplicitTriggerIdList(Oid relationId)
 		 * internal. Hence, below we discard citus_truncate_trigger as well as
 		 * the implicit triggers created by postgres for foreign key validation.
 		 */
+#if PG_VERSION_NUM >= PG_VERSION_15
+
+		/*
+		 * Pre PG15, tgisinternal is true for a "child" trigger on a partition
+		 * cloned from the trigger on the parent.
+		 * In PG15, tgisinternal is false in that case. However, we don't want to
+		 * create this trigger on the partition since it will create a conflict
+		 * when we try to attach the partition to the parent table:
+		 * ERROR: trigger "..." for relation "{partition_name}" already exists
+		 */
```

Comment:
I'm mostly wondering about existing such triggers on upgrades to PG 15.

Say, on PG 14 you have partitioned tables with triggers defined (and also maybe foreign keys as they create internal triggers -- as mentioned on f4566345cf40b068368cb5617e61318da60676ec).

And, you upgrade to PG 15. Can you still do:

- Create a new partition
- Rename / DROP existing triggers  before creating a new partition
- Rename / DROP existing triggers after creating a new partition

I think it'd be great to have such an upgrade test on our TODO list. We don't have to prioritize adding such test now, but seems could have some edge cases here.

Reply:
> Say, on PG 14 you have partitioned tables with triggers defined (and also maybe foreign keys as they create internal triggers -- as mentioned on f4566345cf40b068368cb5617e61318da60676ec).
And, you upgrade to PG 15. Can you still do:
> * Create a new partition
> * Rename / DROP existing triggers before creating a new partition
> * Rename / DROP existing triggers after creating a new partitionThis change doesn't change the behavior of those actions

This change shouldn't change the behavior of those actions. If they were working before, they should work now. If they weren't working before, they shouldn't work now.
I mean, we were skipping triggers for partitions before because they were marked as internal. This change simply and only makes sure we still skip triggers for partitions.

Reply:
makes sense, so should be trivial to add the tests before/after we merge